e2e_test.go

```diff
@@ -0,0 +1,308 @@
+package e2etest
+
+import (
+	"context"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+
+	"github.com/openconfig/clio/collector"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"go.opentelemetry.io/collector/otelcol"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	metricTimestamp = time.Date(2020, 2, 11, 20, 26, 12, 321, time.UTC)
+)
+
+// newTs advances the metricTimestamp and returns a new timestamp.
+func newTs() pcommon.Timestamp {
+	metricTimestamp = metricTimestamp.Add(time.Minute)
+	return pcommon.NewTimestampFromTime(metricTimestamp)
+}
+
+// configProviderSettings is a convenience function to create ConfigProviderSettings that use the
+// local config.yaml.
+func configProviderSettings() otelcol.ConfigProviderSettings {
+	return otelcol.ConfigProviderSettings{
+		ResolverSettings: confmap.ResolverSettings{
+			URIs: []string{filepath.Join("config", "config.yaml")},
+			ProviderFactories: []confmap.ProviderFactory{
+				fileprovider.NewFactory(),
+			},
+		},
+	}
+}
+
+// subscribeRequestForTarget is a convenience function to create a target-specific SubscribeRequest.
+func subscribeRequestForTarget(t *testing.T, target string) *gpb.SubscribeRequest {
```

Comment:
Try adjusting the Subscription mode https://github.com/openconfig/gnmi/blob/900c036ab0c17a4aa6547f173dfb33d367c6f57d/proto/gnmi/gnmi.proto#L302 to avoid coalesing

Reply:
I tried all the different modes and flags, but couldn't get it to not coalesce.